lib/perimeterx/internal/perimeter_x_context.rb

```diff
@@ -43,16 +44,30 @@ def initialize(px_config, req)
       @context[:format] = req.format.symbol
       @context[:score] = 0
 
-      if px_config.key?(:custom_user_ip)
-        @context[:ip] = req.headers[px_config[:custom_user_ip]]
-      elsif px_config.key?(:px_custom_user_ip_method)
+      # Take ip from custom method, if not try from ip headers
+      if px_config.key?(:px_custom_user_ip_method)
```

Comment:
The header list should take precedence over the custom function, since it can be applied with the configuration service